sql: add some tests for lookup join when eq cols are key

yuzefovich opened the pull request:
These tests are "regression tests" for #108489 if it were to be implemented.

Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111173)
<!-- Reviewable:end -->

yuzefovich committed:
sql: add some tests for lookup join when eq cols are key

These tests are "regression tests" for #108489 if it were to be
implemented.

Epic: None

Release note: None

michae2 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111173#-:-NfStGRi4-PkkpKGdH6z:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
